src/os/bluestore/BlueFS.cc

```diff
@@ -2066,6 +2091,138 @@ int BlueFS::device_migrate_to_new(
   return 0;
 }
 
+int BlueFS::migrate_wal_to_v1() {
+  string wal_dir("db.wal");
+  auto dir_it = nodes.dir_map.find(wal_dir);
+  if (dir_it == nodes.dir_map.end()) {
+    dout(5) << fmt::format("{} No files needed to move", __func__) << dendl;
+    return 0;
+  }
+  
+  DirRef dir = dir_it->second;
+  
+  auto file_it = dir->file_map.begin();
+  size_t number_of_wal_v2 = 0;
+  size_t wal_v2_aggregated_size = 0;
+  std::vector<std::pair<string, FileRef>> files_to_migrate;
+  files_to_migrate.reserve(dir->file_map.size());
+  
+  // get info about wal files
+  while(file_it != dir->file_map.end()) {
+    string file_name = file_it->first;
+    FileRef file = file_it->second;
+    if(file->is_new_wal()) {
+      number_of_wal_v2++;
+      wal_v2_aggregated_size += file->fnode.wal_size;
+      files_to_migrate.push_back({ file_name, file });
+    }
+    file_it++;
+  }
+  
+  // check free space
+  uint8_t wal_bdev = vselector->select_prefer_bdev(vselector->get_hint_by_dir("db.wal"));
+  uint64_t free_space = alloc[wal_bdev]->get_free();
+  if (free_space < wal_v2_aggregated_size) {
+    dout(1) << fmt::format("{} Error trying to migrate new WAL V2 to V1. Aggregated size of WAL files is {}, free space is {}", __func__, wal_v2_aggregated_size, free_space) << dendl;
+    return -ENOSPC;
+  }
+  
+  // do migration
+  size_t file_migrated_count = 0;
+  string tmp_name_prefix("__tmp_name_");
```

Comment:
Procedure:
1) convert v2 file "file_name" to v1 file "fixed_temp_name"
2) rename v1 file "fixed_temp_name" to v2 file "file_name"

Wouldn't it be simpler? Are there some conditions that prevent it?

Reply:
I think it is easier to treat errors following my approach. If an error occurred while moving data then you just remove tmp file and leave original one as it was.